pkg/sypgp/sypgp.go

```diff
@@ -636,17 +638,32 @@ func formatMROutput(mrString string, longOutput bool) (int, []byte, error) {
 				if err != nil {
 					return -1, nil, err
 				}
-				keyDateCreated := date(fields[4])
-				keyDateExpired := date(fields[5])
+				// fields 4 through 6 are optional;
+				// they may be missing or empty
+				keyDateCreated := ""
+				if len(fields) >= 4 && fields[4] != "" {
+					keyDateCreated = date(fields[4])
+				}
+
+				keyDateExpired := ""
+				if len(fields) >= 5 && fields[5] != "" {
+					keyDateExpired = date(fields[5])
+				}
 
 				keyStatus := ""
-				if fields[6] == "r" {
-					keyStatus = "[revoked]"
-				} else if fields[6] == "d" {
-					keyStatus = "[disabled]"
-				} else if fields[6] == "e" {
-					keyStatus = "[expired]"
-				} else {
+				if len(fields) >= 6 {
+					flags := fields[6]
```

Comment:
@DrDaveD I believe this should have `strings.ToLower`

Reply:
I didn't see anything the spec allowing it to be uppercase, did you?

Reply:
I don't see anything in the spec allowing for characters other than `r`, `d`, and `e`. So i think we are ok here